activerecord/test/cases/relations_test.rb

```diff
@@ -31,6 +31,40 @@ class TopicWithCallbacks < ActiveRecord::Base
     before_update { |topic| topic.author_name = "David" if topic.author_name.blank? }
   end
 
+  def test_load_associations
+    posts = Post.all
+    even_posts, odd_posts = posts.partition.each_with_index{ |p, i| i.even? }
+
+    assert_queries(1) do
```

Comment:
I was originally thinking that 1) is more of a Preloader testing concern since this new query method doesn't explicitly check whether or not the array's associations are already loaded. Then I did a search--am I right that Preloaded doesn't have any dedicated tests because it's not part of the external API?

Reply:
Exactly. Similar behaviour is probably (hopefully? :smile:) tested via `includes` and friends, but as this is the first time we've provided an API to initiate a preload on demand, it's also the first time we've truly cared how it will behave for a second call. (`includes` could've avoided the problem by working harder to de-dup its values, for example, leaving the Preloader free to consider a redundant load an API error.)

There are places where we directly test behaviour of non-public API, but as a general rule it's safer to cover it explicitly -- that way we're free to refactor/change our internals, assured that we haven't unknowingly affected anything user-visible.